### PR TITLE
Disable AVX register preservation on x86

### PR DESCRIPTION
**Pull request by BradleyWood**
Issue: #15716

Signed-off-by: BradleyWood <bradley.wood@ibm.com>

### CODE REVIEW

**Comment by 0xdaryl** [comment]
Jenkins test sanity xlinux,win,osx jdk11,jdk17

**Comment by gacholio** [comment]
I think it would be preferrable to just add `&& 0`to the ifdef and a comment.

**Comment by 0xdaryl** [comment]
> I think it would be preferrable to just add && 0 to the ifdef and a comment.

Agree.  @BradleyWood, would you do the honours?

File: runtime/vm/jvminit.c [hunk 1]
```diff
@@ -1074,7 +1074,7 @@ initializeJavaVM(void * osMainThread, J9JavaVM ** vmPtr, J9CreateJavaVMParams *c
 	} else if (j9sysinfo_processor_has_feature(&desc, J9PORT_X86_FEATURE_AVX)) {
 		vm->extendedRuntimeFlags |= J9_EXTENDED_RUNTIME_USE_VECTOR_REGISTERS;
 	}
-#endif /* defined(J9HAMMER) && (JAVA_SPEC_VERSION >= 17) */
+#endif /* defined(J9HAMMER) && (JAVA_SPEC_VERSION >= 17) && 0*/
```
Comment (gacholio) [inline runtime/vm/jvminit.c:1077]: Missing space before end comment. Also please add a comment before the ifdef about this being disabled for performance reasons.

File: runtime/vm/jvminit.c [hunk 2]
```diff
@@ -1062,7 +1062,9 @@ initializeJavaVM(void * osMainThread, J9JavaVM ** vmPtr, J9CreateJavaVMParams *c
 	}
 #endif /* J9VM_OPT_JITSERVER */
 
-#if defined(J9HAMMER) && (JAVA_SPEC_VERSION >= 17)
+// Disable AVX+ vector register preservation on x86 due to a
```
Comment (gacholio) [inline runtime/vm/jvminit.c:1065]: Cant use // in a .c file.

**Review by gacholio — APPROVED** [review]

**Comment by 0xdaryl** [comment]
Jenkins test sanity xlinux,win,osx jdk11,jdk17

**Comment by 0xdaryl** [comment]
The Windows build and sanity test appear to have passed and failed after the test run for an infra reason.

**Review by 0xdaryl — APPROVED** [review]

**Pull request merged**